clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/handlers/LoadBalancerV2UpsertHandler.groovy

```diff
@@ -42,46 +51,81 @@ class LoadBalancerV2UpsertHandler {
     return modifyTargetGroupAttributes(loadBalancing, loadBalancer, targetGroup, attributes, null)
   }
   private static String modifyTargetGroupAttributes(AmazonElasticLoadBalancing loadBalancing, LoadBalancer loadBalancer, TargetGroup targetGroup, UpsertAmazonLoadBalancerV2Description.Attributes attributes, DeployDefaults deployDefaults) {
+
     def targetGroupAttributes = []
-    if (attributes) {
-      if (TargetTypeEnum.Lambda.toString().equalsIgnoreCase(targetGroup.getTargetType()))
-      {
-        if (attributes.multiValueHeadersEnabled != null) {
-          targetGroupAttributes.add(new TargetGroupAttribute(key: "lambda.multi_value_headers.enabled", value: attributes.multiValueHeadersEnabled))
-        }
-      } else {
-        Integer deregistrationDelay = [attributes.deregistrationDelay, deployDefaults?.loadBalancing?.deregistrationDelayDefault].findResult(Closure.IDENTITY)
-        if (deregistrationDelay != null) {
-          targetGroupAttributes.add(new TargetGroupAttribute(key: "deregistration_delay.timeout_seconds", value: deregistrationDelay.toString()))
-        }
-        if (loadBalancer.type == 'application') {
-          if (attributes.stickinessEnabled != null) {
-            targetGroupAttributes.add(new TargetGroupAttribute(key: "stickiness.enabled", value: attributes.stickinessEnabled.toString()))
+
+    //Create Target Group Attributes with values provided in description, set to defaults other wise
+    if(deployDefaults !=null) {
```

Comment:
The check on `deployDefaults` and the subsequent `else` branch isn't necessary?

You have already removed default values from `UpsertAmazonLoadBalancerV2Description.TargetGroup.Attributes` , shouldn't we just be able to ensure  that those values are accounted for here?

Reply:
I think the conditional makes sense to account for a create vs an upsert

During a create, we want to supply the default value for all the unspecified attributes
During an upsert, we only want to modify an attribute if it was specifically part of the request, and anything that is null should not be modified

Reply:
Yeah had a chat with @aravindmd this morning. I agree with having a check here but not necessarily `deployDefaults` since it has the potential to be misleading to someone looking at this in the future. I suggested either passing a boolean or split the method into something that would go through the create path and another through the update. It's more of an esthetics opinion. If you don't share that concern, we can keep the check on deplyDefaults as the differentiator between create and update.

Reply:
changed the call to two different methods to make it clear